log/logger.go

```diff
@@ -89,8 +88,7 @@ func (l *Logger) Logf(level logrus.Level, category string, msg string, args ...a
 		return
 	}
 	if l.Logger == nil {
```

Comment:
Let's remove this completely -- `l.Logger` cannot be `nil` at this point, since the actions before it would have failed will `nil` pointer errors.

Reply:
I think the proper action here would be to, in the constructor [here](https://github.com/grafana/xk6-browser/blob/364f7716220ecf3c36ac4a25be65eaf6fb0e8aa3/log/logger.go#L35), provide a "default" logrus Logger in case the input one is `nil`. And then we can remove this part of the code. What do you think?

Reply:
Done in: [61609ee](https://github.com/grafana/xk6-browser/pull/734/commits/61609ee2ccd55f921b99b83ca59228cdadd64d81).

Reply:
@ka3de doing so makes more sense 👍

Reply:
Could we log a warning too? just so that we're aware that we've forgotten to do something?

Reply:
I think that makes sense! although here I see 2 options:
Comply with our logging API to produce the log:
```
// New creates a new logger.
func New(logger *logrus.Logger, iterID string) *Logger {
	var defLogger bool
	if logger == nil {
		defLogger = true
		logger = logrus.New()
	}
	l := &Logger{
		Logger: logger,
		iterID: iterID,
	}
	if defLogger {
		l.Warnf("Logger", "no logger supplied, using default")
	}

	return l
}
```
Or log directly with default logrus logger:
```
// New creates a new logger.
func New(logger *logrus.Logger, iterID string) *Logger {
	if logger == nil {
		logger = logrus.New()
		logger.Warn("no logger supplied, using default")
	}
	return &Logger{
		Logger: logger,
		iterID: iterID,
	}
}
```

What do you think?

Reply:
My vote would be 1, but i'd be happy with either.

Reply:
My vote is 2 since it's simple—1 is complicated.

> Debugging is twice as hard as writing the code in the first place. Therefore, if you write the code as cleverly as possible, you are, by definition, not smart enough to debug it." - Brian W. Kernighan

Reply:
Ops, just pushed option 1 as I thought it was better to comply with our log output, but we can discuss!
The log message itself in this case I don't think it's a big deal, but at the same time I don't think the implementation for the first option is so cumbersome.

Reply:
Less code is better than more code. But, 🤷 , I leave it to you. I said what I wanted to say :)